pkg/asset/installconfig/aws/validation.go

```diff
@@ -211,12 +212,39 @@ func validateDuplicateSubnetZones(fldPath *field.Path, subnets map[string]Subnet
 }
 
 func validateServiceEndpoints(fldPath *field.Path, region string, services []awstypes.ServiceEndpoint) error {
+	var errs []error
+	// For each provided service endpoint, verify we can resolve and connect with net.Dial.
+	for _, service := range services {
+		// Ignore e2e.local from unit tests.
+		if service.URL == "e2e.local" {
```

Comment:
Can we run a server in the unit tests that can be used as a fake service endpoint? That would have a couple benefits.
1) We won't have to have extra code in the production code to deal with special test cases.
2) We can add test cases that verify that the service endpoint connectivity validation is working correctly.

Reply:
This is proving a difficult task. As we discussed today, running a server would be great if we were testing if net.Dial works as expected, but does little to ensure the code here is doing what we expect. Furthermore, we really don't want to be doing external connections for things like DNS. When looking to use gomock, I found it is actually a lot more code because we would need to pass an object into the platform validation function for platforms in order to mock the net functionality. Probably not what we would want either. I'm advocating to continue ignoring net.dial validations for a specified URL, be it `e2e.local` or something more generic such as `localhost`.

Reply:
I think in this case we should be using [`httptest`](https://golang.org/pkg/net/http/httptest/) rather than `gomock`. You can pass the URLs for the test servers in the `ServiceEndpoint`s and assert that the behavior of the function matches the responses of test servers, i.e. returns errors or passes as expected.

Reply:
Let's put it in as is. We can improve up on later if we have a good idea on how to do it easily. As you told me earlier today, the worst that happens with this logic in the function is that we do not fail an installation early that incorrectly uses "e2e.local" as a service endpoint. I can live with that.